src/m365/spo/commands/file/file-list.ts

```diff
@@ -70,46 +81,113 @@ class SpoFileListCommand extends SpoCommand {
     }
 
     try {
-      const files = await this.getFiles(args.options.folder, args);
-      logger.log(files.value);
+      // If --recursive option is specified, retrieve both Files and Folder details, otherwise only Files.
+      const subfolderFiles: FileProperties[] = [];
+      if (args.options.recursive) {
+        const subFolders = await this.getSubfoldersPages(args.options.folder, args, 0);
+
+        for (const folder of subFolders) {
+          const subfolderFilesForFolder: FilePropertiesCollection = await this.getFiles(folder, args, 0);
+          subfolderFilesForFolder.value.forEach((file: FileProperties) => subfolderFiles.push(file));
+        }
+      }
+
+      const files = await this.getFiles(args.options.folder, args, 0);
```

Comment:
If the `skip` parameter has a default value of 0 you don't need to add the value 0 here.